File: AnimalDetails/app/src/main/java/com/example/animaldetails/AnimalViewHolder.java
package com.example.animaldetails;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

public class AnimalViewHolder extends RecyclerView.ViewHolder {
    private TextView tvName;
    private TextView tvSound;
    private ImageView imgAnimal;
    private ClickListener clickListener;
    private CardView cardView;


    public AnimalViewHolder(@NonNull View itemView, ClickListener clickListener) {
        super(itemView);
        this.clickListener = clickListener;
        initViews(itemView);
    }
    private void initViews(View view){
        tvName = view.findViewById(R.id.tvName);
        tvSound = view.findViewById(R.id.tvSound);
        cardView = view.findViewById(R.id.crdView);
        imgAnimal = view.findViewById(R.id.imgAnimal);
    }
    public void setData(AnimalModel model){
        imgAnimal.setImageResource(model.getImgID());
        tvName.setText(model.getTvName());
        tvSound.setText(model.getTvSound());
        cardView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                clickListener.onClick(getAdapterPosition(), model);
            }
        });
    }
}

File: Kotlin Projects/AnimalsprofileinaFragment/app/src/main/java/com/example/animalsprofileinafragment/AnimalViewHolder.kt
package com.example.animalsprofileinafragment

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.animal_item_layout.view.*

class AnimalViewHolder(private val view: View) : RecyclerView.ViewHolder(view) {

    fun setAnimalData(animalModel: AnimalModel) {
        view.apply {
            tvName.text = animalModel.name
            Glide.with(ivProfile).load(animalModel.imagePath).into(ivProfile)
        }
    }
}